src/pages/Settings.js

```diff
@@ -1,53 +1,135 @@
+  majors: { plural: "Majors", singular: "Major" },
+  minors: { plural: "Minors", singular: "Minor" },
+};
+
+const INITIAL_STATE = {
+  specialisations: ["Software Engineering", "Computer Security"],
+  majors: [],
+  minors: ["Economics"],
+};
+
 const Settings = () => {
-  const degreeInfo = {
-    specialisations: {
-      label: "Focus areas/Specialisations",
-      items: ["Software Engineering", "Computer Security"],
-    },
-    majors: { label: "Majors", items: [] },
-    minors: { label: "Minors", items: ["Economics"] },
+  const [mode, setMode] = useState("view");
+  const [degreeInfo, setDegreeInfo] = useState(INITIAL_STATE);
+  const [prevDegreeInfo, setPrevDegreeInfo] = useState({});
+  const [selectionType, setSelectionType] = useState("");
+
+  const handleStartEditMode = () => {
+    setMode("edit");
+    setPrevDegreeInfo(degreeInfo);
   };
+  const handleCancel = () => {
+    setMode("view");
+    setDegreeInfo(prevDegreeInfo);
+  };
+  const handleSaveChanges = () => setMode("view");
 
-  const handleDelete = () => console.log("delete clicked");
-  const handleAdd = () => console.log("add clicked");
-
-  return (
-    <Container id="root" className="d-flex flex-column justify-content-center align-items-center">
-      <Row id="row" className="d-flex justify-content-end page-actions">
-        <Button variant="outline-primary">Cancel</Button>
-        <Button variant="outline-primary">Save Changes</Button>
-      </Row>
-      <Row id="row" className="d-flex justify-content-center page-body">
-        <div className="degree-info">
-          <Row className="d-flex justify-content-end">
-            <span className="info-label">Degree program</span>
-            <span className="info-content">BComp in Computer Science (Hons)</span>
-          </Row>
-          {Object.values(degreeInfo).map(({ label, items }) => (
-            <Row className="d-flex justify-content-end" key={label}>
-              <span className="info-label list-label">{label}</span>
-              <EditableList
-                items={items}
-                onDelete={handleDelete}
-                onAdd={handleAdd}
-                addLabel={`Add ${label.toLowerCase()}`}
-                className="info-content"
-              />
+  const handleOpenSelectionModal = (category) => {
+    setSelectionType(category);
+  };
+
+  const handleAdd = (category, itemToAdd) => {
+    setDegreeInfo({
+      ...degreeInfo,
+      [category]: [...degreeInfo[category], itemToAdd].sort(),
+    });
+    setSelectionType("");
+  };
+
+  const handleDelete = (category, itemToDelete) => {
+    setDegreeInfo({
+      ...degreeInfo,
+      [category]: degreeInfo[category].filter((item) => item !== itemToDelete),
+    });
+  };
+
+  const getOptions = (category) => {
+    if (!category) return [];
+    const existing = new Set(degreeInfo[category]);
+    return selectionOptions[category].filter((option) => !existing.has(option));
+  };
+
+  if (mode === "view") {
```

Comment:
is there a way to return certain parts based on some conditional (instead of 2 whole different chunks for different conditional branches)

JUST WONDERING KEK cos I probably need to do it too

Reply:
you could probably use a ternary operator to render certain parts of the component conditionally